### PR TITLE
documented better developer experience

### PR DESCRIPTION
**Pull request by softprops**
*Issue #, if available:*

*Description of changes:*

It's been noted in a few [open](https://github.com/awslabs/aws-lambda-rust-runtime/issues/25) [issues](https://github.com/awslabs/aws-lambda-rust-runtime/issues/17) that the developer experience for getting a basic rust function built and deployed is less that pleasant. This change incorporates documentation for a less manual workflow using the serverless framework for build/deployment quality life enhancements.

I split off deployment into its own doc section and created a subsection for manual building and setting up a local host and a subsection for using the serverless framework rust plugin which leverages the lambdaci project's provided docker images which faithfully reproduce the required build environment and wires some components together for a more seemless DX for those familiar with serverless framework workflows.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -53,7 +53,13 @@ fn my_handler(event: GreetingEvent, ctx: Context) -> Result<GreetingResponse, Ha
 }
 ```
 
-The code above is the same as the [basic example](https://github.com/awslabs/aws-lambda-rust-runtime/tree/master/lambda-runtime/examples/basic.rs) in the `lambda-runtime` crate. To deploy the basic sample as a Lambda function, we first build it with `cargo`. Since Lambda uses Amazon Linux, you'll need to target your executable for an `x86_64-linux` platform.
+The code above is the same as the [basic example](https://github.com/awslabs/aws-lambda-rust-runtime/tree/master/lambda-runtime/examples/basic.rs) in the `lambda-runtime` crate.
+
+### deployment
```
Comment (davidbarsky) [inline README.md]: Not strictly related to this PR, but I'm thinking we should titlecase all section headers.

File: README.md [hunk 2]
```diff
@@ -86,6 +92,38 @@ $ aws lambda invoke --function-name rustTest \
 $ cat output.json  # Prints: {"message":"Hello, world!"}
 ```
 
+#### serverless framework
+
+You can optionally skip all of the manual building, renaming, and deploying steps above and use the [Serverless framework Rust plugin](https://github.com/softprops/serverless-rust). You can find an example
```
Comment (davidbarsky) [inline README.md]: I think legal/PR would prefer that phrasing would be along the lines of:

“There are currently ways of building this package: manually, and the Serverless framework.“

[manual steps]

“Alternatively, you can build a a Rust-based Lambda function using the Serverless framework Rust plugin”. I _really_ like the Serverless framework, but the phrasing reads (to me, at least) as less of a value judgement.

File: README.md (hunk 1 above)
Comment (davidbarsky) [inline README.md]: Not strictly related to this PR, but I'm thinking we should title-case all section headers.

File: README.md [hunk 3]
```diff
@@ -86,6 +92,38 @@ $ aws lambda invoke --function-name rustTest \
...
+```
+
+Deploy it using the standard serverless workflow
+
+```bash
+# build, package, and deploy service to aws lambda
+$ npx serverless deploy
+```
+
+Invoke it using serverless framework or a configured AWS integrated trigger source
```
Comment (davidbarsky) [inline README.md]: A colon after “source“, please.

**Comment by softprops** [comment]
pushed copy changes

**Pull request merged**